src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -148,6 +150,42 @@ public void visitJavadocToken(DetailNode ast) {
     }
 
     /**
+     * Finds if inheritDoc is placed properly in java doc.
+     * @param ast Javadoc root node.
+     * @return true if inheritDoc is valid or false.
+     */
+    private static boolean isValidInheritDoc(DetailNode ast) {
+        final Set<Integer> allowedTypes = new HashSet<>();
+        allowedTypes.add(JavadocTokenTypes.NEWLINE);
+        allowedTypes.add(JavadocTokenTypes.LEADING_ASTERISK);
+        allowedTypes.add(JavadocTokenTypes.EOF);
```

Comment:
please move creation of this Set to class field, you do not need to create it each time this method is called.